server/datastore/mysql/apple_mdm_test.go

```diff
@@ -71,6 +71,67 @@ func TestIngestMDMAppleDevicesFromDEPSync(t *testing.T) {
 	require.ElementsMatch(t, wantSerials, gotSerials)
 }
 
+func TestDEPSyncTeamAssignment(t *testing.T) {
+	ds := CreateMySQLDS(t)
+	ctx := context.Background()
+	createBuiltinLabels(t, ds)
+
+	depDevices := []godep.Device{
+		{SerialNumber: "abc", Model: "MacBook Pro", OS: "OSX", OpType: "added"},
+		{SerialNumber: "def", Model: "MacBook Pro", OS: "OSX", OpType: "added"},
+	}
+
+	n, err := ds.IngestMDMAppleDevicesFromDEPSync(ctx, depDevices)
+	require.NoError(t, err)
+	require.Equal(t, int64(2), n)
+
+	hosts := listHostsCheckCount(t, ds, fleet.TeamFilter{User: test.UserAdmin}, fleet.HostListOptions{}, 2)
+	for _, h := range hosts {
+		require.Nil(t, h.TeamID)
+	}
+
+	// create a team
+	team, err := ds.NewTeam(ctx, &fleet.Team{Name: "test team"})
+	require.NoError(t, err)
+
+	// assign the team as the default team for DEP devices
+	ac, err := ds.AppConfig(context.Background())
+	require.NoError(t, err)
+	ac.MDM.AppleBMDefaultTeam = team.Name
+	err = ds.SaveAppConfig(context.Background(), ac)
+	require.NoError(t, err)
+
+	depDevices = []godep.Device{
+		{SerialNumber: "abc", Model: "MacBook Pro", OS: "OSX", OpType: "added"},
+		{SerialNumber: "xyz", Model: "MacBook Pro", OS: "OSX", OpType: "added"},
+	}
+
+	n, err = ds.IngestMDMAppleDevicesFromDEPSync(ctx, depDevices)
+	require.NoError(t, err)
+	require.Equal(t, int64(1), n)
+
+	hosts = listHostsCheckCount(t, ds, fleet.TeamFilter{User: test.UserAdmin}, fleet.HostListOptions{}, 3)
+	for _, h := range hosts {
+		if h.HardwareSerial == "xyz" {
+			require.EqualValues(t, team.ID, *h.TeamID)
+		} else {
+			require.Nil(t, h.TeamID)
+		}
+	}
+
+	ac.MDM.AppleBMDefaultTeam = "non-existent"
+	err = ds.SaveAppConfig(context.Background(), ac)
+	require.NoError(t, err)
+
+	depDevices = []godep.Device{
+		{SerialNumber: "jqk", Model: "MacBook Pro", OS: "OSX", OpType: "added"},
+	}
+
+	n, err = ds.IngestMDMAppleDevicesFromDEPSync(ctx, depDevices)
```

Comment:
If I understand correctly, this means that a non-existent team name in the appconfig would prevent any DEP enrollment? I wonder if it'd be better to enroll without a team vs fail to enroll in that scenario (e.g. someone renamed the team and now all new enrollments fail)? (Apologies if that has already been discussed, I may have missed it)

Reply:
great point, AFAIK we haven't discussed it and I agree with you: it would be better to enroll without a team.

@noahtalerman would you mind sanity checking us here? TL;DR: when we sync devices from ABM, how would you like to handle the case when for any reason we can't find the team assigned to them in the config?

1. Assign them to "no team"
2. Fail and don't enroll the devices

Reply:
> it would be better to enroll without a team.

Yes this^ 💯 

@roperzh @mna I prefer assign them to "no team."

Reply:
went ahead and did this in [eb685b9](https://github.com/fleetdm/fleet/pull/9135/commits/eb685b9be3c0a5c930603b5b38b842105aa28120)